srtcore/crypto.cpp

```diff
@@ -449,7 +449,7 @@ void srt::CCryptoControl::sendKeysToPeer(Whether2RegenKm regen SRT_ATR_UNUSED)
                 HLOGC(cnlog.Debug, log << "sendKeysToPeer: SENDING ki=" << ki << " len=" << m_SndKmMsg[ki].MsgLen
                         << " retry(updated)=" << m_SndKmMsg[ki].iPeerRetry);
                 m_SndKmLastTime = now;
-                m_parent->sendSrtMsg(SRT_CMD_KMREQ, (uint32_t *)m_SndKmMsg[ki].Msg, m_SndKmMsg[ki].MsgLen / sizeof(uint32_t));
+                sock->sendSrtMsg(SRT_CMD_KMREQ, (uint32_t *)m_SndKmMsg[ki].Msg, m_SndKmMsg[ki].MsgLen / sizeof(uint32_t));
```

Comment:
Nice, but I think it's not enough.

The `sendKeysToPeer` function should be replaced by something else that checks conditions and fills in a vector that will provide (by being passed by reference) the keys to send, and return bool whether sending is necessary. Regeneration could be this way called from the same place as this function. Not sure yet why `regenCryptoKm` requires `CUDT` when writing this.

Reply:
Simply, the call to `sock->sendSrtMsg` could be replaced by `w_keystosend.push_back(make_pair(...Msg, ...MsgLen/sizeof)...`. The caller of the overall function would check if the result is true, and if so, the keys would be sent through `sendSrtMsg`. This should allow to cut off also the `CCryptoControl` class the dependency on `CUDT` details.

Reply:
This is the minimum set of changes I need to be able to write unit tests around `CCryptoControl`.
I don't like using `std::vector` as it would involve dynamic memory allocation. I should be enough to return an array of two pointers to the KM structures already stored by `CCRyptoLibrary`.